### DISCUSSION TITLE
Error 504 and "An error occurred with your deployment" message when calling APIs after deployment on Vercel

### DISCUSSION THREAD

**Discussion by ravindrasirvi609** [vercel/next.js#52655] [Help] [answered]
### Summary

I have encountered an issue with my project deployed on Vercel, specifically when calling my APIs. Upon making API requests, I consistently receive a status code 504 and an error message stating "An error occurred with your deployment." However, when running the project locally, the APIs function properly without any errors.

I have thoroughly tested the project on my local environment and confirmed that all APIs are working as expected. The issue only arises after deploying the project on Vercel.

This problem obstructs my application's smooth functioning

, as it relies on these APIs to perform crucial tasks. I am seeking assistance from the Vercel community to help identify and resolve this issue.

Please let me know if any additional information or code snippets are required to better understand the problem. Your guidance and support in resolving this issue would be greatly appreciated.

https://operantpharmacy.vercel.app/signup

<img width="1440" alt="Screenshot 2023-07-13 at 9 49 30 PM" src="https://github.com/vercel/next.js/assets/84620599/baeb76bf-79e3-48e6-938e-358d8924ddc1">
<img width="1440" alt="Screenshot 2023-07-13 at 9 49 43 PM" src="https://github.com/vercel/next.js/assets/84620599/54cf153b-dfa7-4621-9ed3-277616954fe9">

### Example

https://operantpharmacy.vercel.app/signup

**Comment by icyJoseph** [comment]
Have you provided `process.env.MONGO_URI` in your Vercel Dashboard?

**Comment by ravindrasirvi609** [comment, marked as answer]
Thanks, @icyJoseph  for Replying

Problem:
I encountered connectivity errors with my MongoDB database because the network access was restricted to only my local IP address.

**Solution**:
After identifying the issue, I modified the MongoDB configuration to allow access from any IP address (0.0.0.0/0).

Result:
With the updated settings, I successfully connected to the database from external sources, resolving the problem.
